Possible sprintf buffer overflow

This can be found by enabling `-Wformat`, I think, or at least one of the following flags anyway:
```
$ gcc -m64 -I/usr/include/R -DNDEBUG -I/usr/include  -I/usr/local/include  -I/usr/include -fpic  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic  -c R_zmq_utility.c -o R_zmq_utility.o
R_zmq_utility.c: In function 'R_zmq_version':
R_zmq_utility.c:34:24: warning: '__builtin___sprintf_chk' writing a terminating nul past the end of the destination [-Wformat-overflow=]
  sprintf(ver, "%d.%d.%d", major, minor, patch);
                        ^
In file included from /usr/include/stdio.h:939:0,
                 from /usr/include/R/R.h:56,
                 from R_zmq.h:4,
                 from R_zmq_utility.c:1:
/usr/include/bits/stdio2.h:33:10: note: '__builtin___sprintf_chk' output between 6 and 36 bytes into a destination of size 5
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Note that this is using the tarball from CRAN where `ver` is 5 bytes long. I see that `master` has increased the size of `ver` to 6, but that is not sufficient. It's pretty unlikely that any version would be a billion, but it's certainly a possibility that `minor=10` or `patch=100`.

PS, please add tags for the releases.

--- a/src/R_zmq_utility.c
+++ b/src/R_zmq_utility.c
@@ -29,15 +29,15 @@ SEXP R_zmq_version(){
 	char ver[36];
 	int chars;
 	SEXP ret;
-	
+
 	zmq_version(&major, &minor, &patch);
 	//Rprintf("Current ZeroMQ version is %d.%d.%d\n", major, minor, patch);
-	
+
 	chars = sprintf(ver, "%d.%d.%d", major, minor, patch);
-	
+
 	ret = PROTECT(allocVector(STRSXP, 1));
 	SET_STRING_ELT(ret, 0, mkCharLen(ver, chars));
-	
+
 	UNPROTECT(1);
 	return(ret);
 } /* End of R_zmq_version(). */

--- a/src/R_zmq_utility.c
+++ b/src/R_zmq_utility.c
@@ -25,16 +25,18 @@ SEXP R_zmq_strerror(SEXP R_errno){
 /* Version. */
 SEXP R_zmq_version(){
 	int major, minor, patch;
-	char ver[6];
+	/* (10 bytes for int + 1 byte for sign) * 3 + 2 dots + 1 NUL */
+	char ver[36];
+	int chars;
 	SEXP ret;
 	
 	zmq_version(&major, &minor, &patch);
 	//Rprintf("Current ZeroMQ version is %d.%d.%d\n", major, minor, patch);
 	
-	sprintf(ver, "%d.%d.%d", major, minor, patch);
+	chars = sprintf(ver, "%d.%d.%d", major, minor, patch);
 	
 	ret = PROTECT(allocVector(STRSXP, 1));
-	SET_STRING_ELT(ret, 0, mkCharLen(ver, 5));
+	SET_STRING_ELT(ret, 0, mkCharLen(ver, chars));
 	
 	UNPROTECT(1);
 	return(ret);

